mdp/nodes/classifier_nodes.py

```diff
@@ -228,24 +292,36 @@ def _train_one(self, pattern):
 
     @property
     def memory_size(self):
-        """Returns the Hopfield net's memory size"""
+        """
+        :return: The Hopfield net's memory size
+        :rtype: int
+        """
         return self.input_dim
 
     @property
     def load_parameter(self):
-        """Returns the load parameter of the Hopfield net.
-        The quality of memory recall for a Hopfield net breaks down when the
-        load parameter is larger than 0.14."""
+        """
+        .. note:: The quality of memory recall for a Hopfield net breaks down when the
+            load parameter is larger than 0.14.
+
+        :return: The load parameter of the Hopfield net.
+        :rtype: float
+        """
         return old_div(self._num_patterns, float(self.input_dim))
 
     def _stop_training(self):
         # remove self-feedback
-        # we could use numx.fill_diagonal, but thats numpy 1.4 only
+        # we could use numx.fill_diagonal, but thats numpy 1. 4 only
         for i in range(self.input_dim):
             self._weight_matrix[i][i] = 0
 
     def _label(self, x, threshold = 0):
         """Retrieves patterns from the associative memory.
+        
+        :param x: A matrix having different variables on different columns
+            and observations on rows.
```

Comment:
why not add ``:type x: numpy.ndarray``?

Reply:
IIRC I was not sure at the time whether MDP generalized to iterables. Due to limited knowledge of the specific modules, I was easy on said types. If that abstraction is not featured for this class, I will adapt the docstring. See http://mdp-toolkit.sourceforge.net/tutorial/iterables.html

Reply:
Okay, I see. We should try to figure out the actual type in these cases.
Yes, often np.ndarray actually means 'arraylike' in the sense "everything that np.array eats without complaint" but thats (hopefully) a distinct issue.
What about ``numpy.ndarray or iterable of numpy.ndarray``?
Here PEP 484 notation might be favorable, i.e. ``Union[numpy.ndarray, Iterable[numpy.ndarray]]``. But then we should use it consistently; not a box I'd like to open at this stage of doc refactoring.

Reply:
As discussed, I have added the `numpy.ndarray` type and a `# todo: consider iterables` comment for each of the cases.